### ISSUE TITLE
problems with setting 10mbit+

### ISSUE THREAD

**Issue by buzzy** [magnific0/wondershaper#2]
Does wondershaper support faster speeds than 9mbit?

When using:

"wondershaper br23 9000 9000"

It works just fine. But settings something like:

"wondershaper br23 100000 100000"

It does not limit the speed to 100mbit (about 12MB/sec).

If I run a "wondershaper br23" I see that 10mbit is hardcoded (?) on the first row:

"qdisc cbq 1: root refcnt 2 rate 10000Kbit (bounded,isolated) prio no-transmit"

Am I doing something wrong?

**Comment by leoetlino** [comment]
There you go! Now only just waiting on the repo owner.

**Comment by dlio** [comment]
Confirming that buzzy's fix (and leoetlino's commit) work.  Great tool; great fix.  Thanks guys.

**Comment by heavyhdx** [comment]
Using this fix doesn't seem to work anymore. Now the download speed just gets completely crippled if you set it higher. Without the fix it simply stays capped at 10 mbit/s.

**Comment by leoetlino** [comment]
I don't need to use Wondershaper anymore, but maybe try PR #9?

**Comment by psycholyzern** [comment]
Anyone know which build is working/ 
I have tried forked version which applied the patch, but my download speed would be less than a few kBps. Only upload limit is working.

**Comment by TruboXL** [comment]
@psycholyzern 
There's an original script (wshaper.htb) using HTB Queuing by the original author...
http://lartc.org/wondershaper/

You can check that out first, but I don't will fix your issue :(

**Comment by T0aD** [comment]
I confirmed its working with the following fix on Ubuntu 15.10:
sed -i 's/10mbit/1000mbit/' /sbin/wondershaper

**Comment by kpreslan** [comment]
I'm still having problems with the fix.  My download seems to be limited to about a 1mbit when it should be going faster than that.  This is on Ubuntu Trusty.

Out of curiosity, does this line look wrong to anyone else?
NOPRIOHOSTSRC=80

Shouldn't port 80 be on the "NOPRIOPORTSRC" line?  I think NOPRIOHOSTSRC wants an address or netmask or something.

**Comment by magnific0** [comment]
I have very belatedly accepted #9 can someone confirm this fixes the issue with setting rates higher than 10mbit?

**Comment by magnific0** [comment]
I have gone ahead and installed wondershaper on my system again (I have no need for it in recent years) and tested the new code. Everything seems to work fine, so I'm closing the issues. If anyone experiences more problems. Let me know.

**Comment by SushantMongia** [comment]
Hi magnific0,
I tried working with the most recent version of wondershaper, oddly it limits the upload speed correctly but cripples the download speed to less than 0.15 Mega Bits persec, every time. Please let me know!
Best
Sushant

**Comment by magnific0** [comment]
@SushantMongia thanks for reporting.

Can you please let me know which kernel you're using:

    uname -a

Also does this behaviour also occur when you only limit the upload (so without the ```-d <dowload_rate>```?

    sudo wondershaper -c -a <adapter>
    sudo wondershaper -a <adapter> -u <upload_rate>

**Comment by magnific0** [comment]
@SushantMongia perhaps it's better to open a separate issue for this. As this thread contains at least 3 unrelated issues already and is getting unclear.

**Comment by ponahoum** [comment]
I confirm this issue with docker containers. Very annoying

wondershaper eth0 245760 8192

Limits downloads to 1mbps instead of 30mbps

**Comment by McDime** [comment]
If you end-up in this thread, as I did, please make sure that you use the latest version of this nice tool. Unfortunately, as of now, the EPEL repo contains still a "problematic" version and the fix described above also did not help in my setup. However, the version from the home page (as of now 1.4) works just fine.

**Comment by beck-weber-ing** [comment]
As of today, for Debian 'Bullseye' this fix is still needed and works.
